src/diff/component.js

```diff
@@ -190,8 +189,17 @@ export function renderComponent(
 		globalContext = assign(assign({}, globalContext), c.getChildContext());
 	}
 
-	if (!isNew && c.getSnapshotBeforeUpdate != null) {
-		snapshot = c.getSnapshotBeforeUpdate(oldProps, oldState);
+	if (!isNew) {
+		if (c.getSnapshotBeforeUpdate != null) {
+			snapshot = c.getSnapshotBeforeUpdate(oldProps, oldState);
+		}
+
+		// Only schedule componentDidUpdate if the component successfully rendered
+		if (c.componentDidUpdate != null) {
+			c._renderCallbacks.push(() => {
+				c.componentDidUpdate(oldProps, oldState, snapshot);
+			});
+		}
```

Comment:
This change is to prevent queueing up extra calls to `cDU` for each aborted render of a suspending component. Since a suspending component is guaranteed to be rendered at least 3 times (1. initial render, 2. re-render of fallback, 3. successfuly re-render when promise resolves), I figured we should not call cDU so many times unnecessarily.